app/src/main/java/com/safetyheads/akademiaandroida/ActivitiesList/FragmentsAdapter.kt

```diff
@@ -0,0 +1,46 @@
+package com.safetyheads.akademiaandroida.ActivitiesList
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.view.isVisible
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentTransaction
+import androidx.recyclerview.widget.RecyclerView
+import com.safetyheads.akademiaandroida.R
+import com.safetyheads.akademiaandroida.databinding.CardBinding
+
+class FragmentsAdapter(private val list: List<Fragment>, private val context: Context) :
+    RecyclerView.Adapter<FragmentsAdapter.MyViewHolder>() {
+
+    class MyViewHolder(binding: CardBinding) : RecyclerView.ViewHolder(binding.root) {
+        val textViewCard = binding.textViewCard
+    }
+
+    override fun onCreateViewHolder(
+        parent: ViewGroup,
+        viewType: Int
+    ): MyViewHolder {
+        val inflater = LayoutInflater.from(parent.context)
+        val cardBinding = CardBinding.inflate(inflater, parent, false)
+        return MyViewHolder(cardBinding)
+    }
+
+    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
+        val itemFragment = list[position]
+        val name = list[position]::class.java.simpleName
+        holder.textViewCard.text = name
+        val fragmentManager = (context as AppCompatActivity).supportFragmentManager
+
+        holder.itemView.setOnClickListener {
+            val fragmentTransaction: FragmentTransaction = fragmentManager.beginTransaction()
```

Comment:
![image](https://user-images.githubusercontent.com/3758285/219418288-0dddeb2d-612d-4f46-81ad-085f8a7b34f5.png)
To nie jest chyba oczekiwany rezultat. Fragment powinien zostać dodany do stack albo zastąpić istniejący.